From 3f80a85fa600028709f617c254b901ecdbf02d1a Mon Sep 17 00:00:00 2001
From: David Disseldorp <ddiss@suse.de>
Date: Mon, 7 Sep 2015 13:08:54 +0200
Subject: [PATCH] target: remove hardcoded T10 Vendor ID in INQUIRY response
Patch-mainline: 5.0-rc1
Git-commit: 2d882847280e3ae1ddc95175d0fc2006e11bb63f
References: fate#318836

Use the value stored in t10_wwn.vendor, which defaults to "LIO-ORG", but
can be reconfigured via the vendor_id ConfigFS attribute.

Signed-off-by: David Disseldorp <ddiss@suse.de>
Reviewed-by: Bryant G. Ly <bly@catalogicsoftware.com>
Reviewed-by: Lee Duncan <lduncan@suse.com>
Reviewed-by: Hannes Reinecke <hare@suse.com>
Reviewed-by: Roman Bolshakov <r.bolshakov@yadro.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>

---
 drivers/target/target_core_spc.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/drivers/target/target_core_spc.c
+++ b/drivers/target/target_core_spc.c
@@ -114,7 +114,8 @@ spc_emulate_inquiry_std(struct se_cmd *c
 	 * unused bytes shall be filled with ASCII space characters (20h).
 	 */
 	memset(&buf[8], 0x20, 8 + 16 + 4);
-	memcpy(&buf[8], "LIO-ORG", sizeof("LIO-ORG") - 1);
+	memcpy(&buf[8], dev->t10_wwn.vendor,
+	       strnlen(dev->t10_wwn.vendor, 8));
 	memcpy(&buf[16], dev->t10_wwn.model,
 	       strnlen(dev->t10_wwn.model, 16));
 	memcpy(&buf[32], dev->t10_wwn.revision,
@@ -258,7 +259,8 @@ check_t10_vend_desc:
 	buf[off+2] = 0x0;
 	/* left align Vendor ID and pad with spaces */
 	memset(&buf[off+4], 0x20, 8);
-	memcpy(&buf[off+4], "LIO-ORG", sizeof("LIO-ORG") - 1);
+	memcpy(&buf[off+4], dev->t10_wwn.vendor,
+	       strnlen(dev->t10_wwn.vendor, 8));
 	/* Extra Byte for NULL Terminator */
 	id_len++;
 	/* Identifier Length */
